app/screens/server/index.tsx

```diff
@@ -317,7 +317,16 @@ const Server = ({
             return;
         }
 
-        const server = await getServerByIdentifier(data.config!.DiagnosticId);
+        if (!data.config?.DiagnosticId) {
+            setUrlError(formatMessage({
+                id: 'mobile.diagnostic_id.empty',
+                defaultMessage: 'The server you are trying to access doesn\'t have a DiagnosticId set. Contact the system admin to check the configuration and make sure the DiagnosticId is set.',
```

Comment:
My only comment about this is that I guess probably this is ok for on prem, but what about cloud (where the customer does not control the database, since diagnosticID is set without user action?)

Reply:
Cloud is up to date, and won't suffer from this (in theory). There are changes in the server code for this to never happen. But since we have to be backwards compatible, this is still valuable.